Add document for launching Ignition node from source and Local network from source

AurelienFT opened the pull request:
## Description

New documentation on README.md that covers more cases to launch nodes on differents cases.

acerone85 commented on README.md:
````diff
@@ -139,19 +141,131 @@ fuel-core run \
+git checkout v0.40.2
+```
+
+Generate a keyparr for your node :
````
```suggestion
Generate a keypair for your node :
```

acerone85 commented on README.md:
```diff
@@ -139,19 +141,131 @@ fuel-core run \
+
+## Running a Local network from source
+
+Clone this repository : 
```
```suggestion
Clone the `fuel-core` repository : 
```

acerone85 reviewed (approved):
Followed the README to launch a local node, everything works as expected.

xgreenx commented on README.md:
```diff
@@ -139,19 +141,131 @@ fuel-core run \
+
+Instead of directly placing your personal values on the command we advise you to use, for example, environment variables.
+
+### From source
```
What is the difference with "Running a Local network from source"?

AurelienFT replied:
The difference is the parameters, here we sue the chain config of ignition we use the realyer address on mainnet, we don't produce blocks, we use the bootstrap peers...

xgreenx commented on README.md:
```diff
@@ -139,19 +141,131 @@ fuel-core run \
+--keypair {KEYGEN_SECRET_KEY} \
+--relayer {ETHEREUM_RPC_ENDPOINT} \
+--ip=0.0.0.0 \
+--port 4000 \
+--peering-port 30333 \
 --db-path ~/.fuel-ignition \
 --snapshot {PATH_TO_CHAIN_CONFIGURATION_FOLDER}/ignition \
---utxo-validation --poa-instant false --enable-p2p \
+--utxo-validation \
+--poa-instant false \
+--enable-p2p \
 --bootstrap-nodes /dnsaddr/mainnet.fuel.network \
 --sync-header-batch-size 100 \
---relayer-v2-listening-contracts=0xAEB0c00D0125A8a788956ade4f4F12Ead9f65DDf \
---relayer-da-deploy-height=20620434 \
---relayer-log-page-size=100 \
---sync-block-stream-buffer-size 30
+--relayer-v2-listening-contracts 0xAEB0c00D0125A8a788956ade4f4F12Ead9f65DDf \
+--relayer-da-deploy-height 20620434 \
+--relayer-log-page-size 2000 \
+--sync-block-stream-buffer-size 30 \
+--min-gas-price 1000 \
+--starting-gas-price 1000 \
+--gas-price-change-percent 10 \
+--gas-price-threshold-percent 50
```
We don't need to provide information which fields to use, for that we have documentation and we can point to it. Maybe we need to rework "Running a Ignition node" and point to the documentation to avoid multiple places.

We just need to describe how to build the production ready binary.

AurelienFT replied:
I don't see where is the documentation. Can you point it out to me ? I think it's valuable to be able to copy-paste a command that is working as expected to connect to Ignition/Launch a local network and then if you need something specific you can just tweak some parameters. We saw in the past weeks that if some parameters are not present the node isn't working as expected so to avoid confusion I find it useful to first set everything correctly (such as a `--ignition` that we could have) and then play with flags if you want.

xgreenx replied:
https://docs.fuel.network/guides/running-a-node/running-a-mainnet-node/

AurelienFT replied:
Updated to remove duplicate

xgreenx commented on README.md:
````diff
@@ -139,19 +141,131 @@ fuel-core run \
+
+Run your node (change all variable with {} to your own personal variables):
+```bash
+cargo run --release --bin fuel-core -- run \
````
You are missing "production" feature. The command should looks like `cargo build --release -p fuel-core-bin --no-default-features --features production`.

AurelienFT replied:
Done

xgreenx reviewed (commented):
I think we need to put instruction how to build binary image between "Rust setup" and "Compiling".

And we can rework "Compiling" section to specify that it is for local development.

AurelienFT commented:
@xgreenx I moved the section after the Rust setup. I think it's valuable to have a section for node runners with pre made commands etc and have the general commands for people that wants to contribute to the node. I made two distinct sections and moved the node running one to the top.
